openapi-core/src/main/java/io/ballerina/openapi/core/generators/schema/ballerinatypegenerators/EnumGenerator.java

```diff
@@ -72,13 +72,17 @@ public TypeDescriptorNode generateTypeDescriptorNode() throws BallerinaOpenApiEx
                     enumBuilder.append(enumValue).append("|");
                 }
             }
+            if (enumBuilder.length() > 0) {
+                enumBuilder.deleteCharAt(enumBuilder.length() - 1);
+                String enumString = isNull ? enumBuilder.toString() + "?" : enumBuilder.toString();
+                return NodeParser.parseTypeDescriptor(enumString);
+            } else {
+                throw new BallerinaOpenApiException(String.format("Enum list of the type '%s' is empty.",
+                        schema.getType()));
+            }
```

Comment:
Should we throw an exception if we find an enum with no elements? Isn't that something which is not allowed in the spec? (incase if its allowed, shouldn't we simply ignore and just create the ballerina type from the given enum type?)

Reply:
I actually added this error handling to address a scenario like below (This is a hypothetical scenario which has not found in any of the openapi files)

```yaml
components:
  schemas:
    MeetingTypes:
      type: string
      enum:
        - null
```

This is a valid openapi syntax and does not give any parser error. Other than this could not come up any other scenario where the enum list is empty. 

Without throwing the error we can generate the above type as below

```bal
public type MeetingTypes string?
```
```

Reply:
I see.. IMO its better if we can generate the Type as `<SCHEMA_TYPE>?` in such cases, given that its a valid openAPI syntax (as per what you mentioned above) instead of throwing an error

Reply:
also lets make sure to modify our test cases to capture this scenario as well